### ISSUE TITLE
Tornado error after installing ipython-sql

### ISSUE THREAD

**Issue by stevenbeeckman** [ipython/ipython#7689]
This issue was first reported in the `ipython-sql` repo catherinedevlin/ipython-sql#29 as the issue manifested itself when trying to install `ipython-sql`. A TL;DR follows below.

As mentioned in that issue, I'm (very) new to IPython and IPython Notebook. Just want to get the SQL integration shown to my boss. I managed to get IPython Notebook running on an Ubuntu with Vagrant and even save 3 notebooks!

Trying to get the `%sql` magic running led me to a path of upgrading `jinja2`, `tornado` etc. Currently I'm stuck at not having a browser connected at run time. Here's the output of my console:

```bash
vagrant@precise32:~$ ipython notebook --ip=0.0.0.0
/usr/local/lib/python2.7/dist-packages/IPython/frontend.py:30: UserWarning: The top-level `frontend` package has been deprecated. All its subpackages have been moved to the top `IPython` level.
  warn("The top-level `frontend` package has been deprecated. "
[I 23:31:13.374 NotebookApp] Using existing profile dir: u'/home/vagrant/.ipython/profile_default'
[I 23:31:13.391 NotebookApp] Using MathJax from CDN: https://cdn.mathjax.org/mathjax/latest/MathJax.js
[I 23:31:13.451 NotebookApp] Serving notebooks from local directory: /home/vagrant
[I 23:31:13.453 NotebookApp] 0 active kernels
[I 23:31:13.455 NotebookApp] The IPython Notebook is running at: http://0.0.0.0:8888/
[I 23:31:13.456 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 23:31:13.457 NotebookApp] No web browser found: could not locate runnable browser.
Traceback (most recent call last):
  File "/usr/bin/ipython", line 8, in <module>
    launch_new_instance()
  File "/usr/local/lib/python2.7/dist-packages/IPython/config/application.py", line 574, in launch_instance
    app.start()
  File "/usr/local/lib/python2.7/dist-packages/IPython/terminal/ipapp.py", line 367, in start
    return self.subapp.start()
  File "/usr/local/lib/python2.7/dist-packages/IPython/html/notebookapp.py", line 1064, in start
    self.io_loop = ioloop.IOLoop.current()
AttributeError: type object 'IOLoop' has no attribute 'current'

If you suspect this is an IPython bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@scipy.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    c.Application.verbose_crash=True

vagrant@precise32:~$ python -c "import IPython; print(IPython.sys_info())"
{'commit_hash': u'ee6223a',
 'commit_source': 'installation',
 'default_encoding': 'ISO-8859-1',
 'ipython_path': '/usr/local/lib/python2.7/dist-packages/IPython',
 'ipython_version': '3.0.0-b1',
 'os_name': 'posix',
 'platform': 'Linux-3.2.0-23-generic-pae-i686-with-Ubuntu-12.04-precise',
 'sys_executable': '/usr/bin/python',
 'sys_platform': 'linux2',
 'sys_version': '2.7.3 (default, Dec 18 2014, 19:03:52) \n[GCC 4.6.3]'}
vagrant@precise32:~$
```

Which leads me to believe there's something wrong with the Tornado usage.

**Comment by stevenbeeckman** [comment]
Here's a gist to reproduce: https://gist.github.com/stevenbeeckman/beffbbd867b93575ac48

**Comment by takluyver** [comment]
What version of tornado do you have? I've tried this with the latest stable tornado (4.0.2) and the latest prerelease (4.1b2) and I don't see an issue with either.

**Comment by minrk** [comment]
also, what pyzmq version?

**Comment by stevenbeeckman** [comment]
TL;DR from `pip freeze`
pyzmq==2.1.11
tornado==4.1b2

Right after `sudo pip install ipython-sql`:

```bash
vagrant@precise32:~$ pip freeze
Warning: cannot find svn location for distribute==0.6.24dev-r0
GnuPGInterface==0.3.2
MySQL-python==1.2.3
PyMySQL==0.6.3
SQLAlchemy==0.9.8
apt-xapian-index==0.44
argparse==1.2.1
chardet==2.0.1
command-not-found==0.2.44
configobj==4.7.2
decorator==3.3.2
## FIXME: could not find svn URL in dependency_links for this package:
distribute==0.6.24dev-r0
ipython==3.0.0-b1
ipython-sql==0.3.5
language-selector==0.1
matplotlib==1.1.1rc
numpy==1.6.1
pexpect==2.3
prettytable==0.7.2
pycurl==7.19.0
pyparsing==1.5.2
python-apt==0.8.3ubuntu7
python-dateutil==1.5
python-debian==0.1.21ubuntu1
pytz==2011k
pyzmq==2.1.11
simplegeneric==0.7
six==1.9.0
sqlparse==0.1.14
tornado==2.1
ufw==0.31.1-1
wsgiref==0.1.2
vagrant@precise32:~$
```

Then running `ipython notebook --ip=0.0.0.0` complains about jinja2, tornado and jsonschema. Installing or upgrading them using `sudo pip install --upgrade jinja2 tornado jsonschema`.

```bash
vagrant@precise32:~$ pip freeze
Warning: cannot find svn location for distribute==0.6.24dev-r0
GnuPGInterface==0.3.2
Jinja2==2.7.3
MarkupSafe==0.23
MySQL-python==1.2.3
PyMySQL==0.6.3
SQLAlchemy==0.9.8
apt-xapian-index==0.44
argparse==1.2.1
backports.ssl-match-hostname==3.4.0.2
certifi==14.05.14
chardet==2.0.1
command-not-found==0.2.44
configobj==4.7.2
decorator==3.3.2
## FIXME: could not find svn URL in dependency_links for this package:
distribute==0.6.24dev-r0
ipython==3.0.0-b1
ipython-sql==0.3.5
jsonschema==2.4.0
language-selector==0.1
matplotlib==1.1.1rc
numpy==1.6.1
pexpect==2.3
prettytable==0.7.2
pycurl==7.19.0
pyparsing==1.5.2
python-apt==0.8.3ubuntu7
python-dateutil==1.5
python-debian==0.1.21ubuntu1
pytz==2011k
pyzmq==2.1.11
simplegeneric==0.7
six==1.9.0
sqlparse==0.1.14
tornado==4.1b2
ufw==0.31.1-1
wsgiref==0.1.2
vagrant@precise32:~$
```

```bash
vagrant@precise32:~$ ipython notebook --ip=0.0.0.0
[I 21:49:39.458 NotebookApp] Using existing profile dir: u'/home/vagrant/.ipython/profile_default'
[I 21:49:39.470 NotebookApp] Writing notebook server cookie secret to /home/vagrant/.ipython/profile_default/security/notebook_cookie_secret
[I 21:49:39.473 NotebookApp] Using MathJax from CDN: https://cdn.mathjax.org/mathjax/latest/MathJax.js
[W 21:49:39.505 NotebookApp] Terminals not available (error was No module named terminado)
[I 21:49:39.506 NotebookApp] Serving notebooks from local directory: /home/vagrant
[I 21:49:39.508 NotebookApp] 0 active kernels
[I 21:49:39.509 NotebookApp] The IPython Notebook is running at: http://0.0.0.0:8888/
[I 21:49:39.510 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 21:49:39.511 NotebookApp] No web browser found: could not locate runnable browser.
Traceback (most recent call last):
  File "/usr/local/bin/ipython", line 9, in <module>
    load_entry_point('ipython==3.0.0-b1', 'console_scripts', 'ipython')()
  File "/usr/local/lib/python2.7/dist-packages/IPython/__init__.py", line 120, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/IPython/config/application.py", line 574, in launch_instance
    app.start()
  File "/usr/local/lib/python2.7/dist-packages/IPython/terminal/ipapp.py", line 367, in start
    return self.subapp.start()
  File "/usr/local/lib/python2.7/dist-packages/IPython/html/notebookapp.py", line 1064, in start
    self.io_loop = ioloop.IOLoop.current()
AttributeError: type object 'IOLoop' has no attribute 'current'

If you suspect this is an IPython bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@scipy.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    c.Application.verbose_crash=True

vagrant@precise32:~$
```

**Comment by takluyver** [comment]
> pyzmq==2.1.11

Try upgrading that.

**Comment by minrk** [comment]
#7694 should give a better error message

**Comment by stevenbeeckman** [comment]
Upgrading pyzmq worked. Thanks a lot! 

I updated my gist so people can get up and running quick: https://gist.github.com/stevenbeeckman/beffbbd867b93575ac48
